bigtable/bigtable-proxy/src/main/java/com/google/cloud/bigtable/examples/proxy/commands/Verify.java

```diff
@@ -0,0 +1,251 @@
+import io.opentelemetry.contrib.gcp.resource.GCPResourceProvider;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.metrics.internal.data.ImmutableGaugeData;
+import io.opentelemetry.sdk.metrics.internal.data.ImmutableLongPointData;
+import io.opentelemetry.sdk.metrics.internal.data.ImmutableMetricData;
+import io.opentelemetry.sdk.resources.Resource;
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Iterator;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Help.Visibility;
+import picocli.CommandLine.Option;
+
+@Command(name = "verify", description = "Verify environment is properly set up")
+public class Verify implements Callable<Void> {
+  @Option(
+      names = "--bigtable-project-id",
+      required = true,
+      description = "Project that contains a Bigtable instance to use for connectivity test")
+  String bigtableProjectId;
+
+  @Option(
+      names = "--bigtable-instance-id",
+      required = true,
+      description = "Bigtable instance to use for connectivity test")
+  String bigtableInstanceId;
+
+  @Option(
+      names = "--bigtable-table-id",
+      required = true,
+      description = "Bigtable table to use for connectivity test")
+  String bigtableTableId;
+
+  @Option(
+      names = "--metrics-project-id",
+      required = true,
+      description = "The project id where metrics should be exported")
+  String metricsProjectId = null;
+
+  @Option(
+      names = "--bigtable-data-endpoint",
+      converter = Endpoint.ArgConverter.class,
+      showDefaultValue = Visibility.ALWAYS)
+  Endpoint dataEndpoint = Endpoint.create("bigtable.googleapis.com", 443);
+
+
+  Credentials credentials = null;
+
+  @Override
+  public Void call() throws Exception {
+    if (credentials == null) {
+      credentials = GoogleCredentials.getApplicationDefault();
+    }
+    checkBigtable(
+        MoreCallCredentials.from(credentials),
+        String.format(
+            "projects/%s/instances/%s/tables/%s",
+            bigtableProjectId, bigtableInstanceId, bigtableTableId));
+
+    checkMetrics(credentials);
+    return null;
+  }
+
+  private void checkBigtable(CallCredentials callCredentials, String tableName) {
+    ManagedChannel channel =
+        ManagedChannelBuilder.forAddress(dataEndpoint.getName(), dataEndpoint.getPort()).build();
+
+    try {
+      Metadata md = new Metadata();
+      PercentEscaper escaper = new PercentEscaper("", true);
+      md.put(
+          Key.of("x-goog-request-params", Metadata.ASCII_STRING_MARSHALLER),
+          String.format("table_name=%s&app_profile_id=%s", escaper.escape(tableName), ""));
+
+      BigtableBlockingStub stub =
```

Comment:
why not using the client directly? I assume it's because we don't need to import the client dependency?

Reply:
yep